### PR TITLE
noarch python: convert to noarch package

### PR DESCRIPTION
**Pull request by oliver-sanders**
closes https://github.com/conda-forge/cylc-uiserver-feedstock/issues/21

Convert to a `noarch: python` package before the RC1 release to avoid disruption.

This project meets the `noarch: python` requirements:

* [x] No compiled extensions
* [x] No post-link or pre-link or pre-unlink scripts
* [x] No OS-specific build scripts
* [x] No python version specific requirements
* [x] No skips except for python version. If the recipe is py3 only, remove skip statement and add version constraint on python in host and run section.
* [x] 2to3 is not used
* [x] Scripts argument in setup.py is not used
* [ ] If console_scripts entry_points are defined in setup.py or setup.cfg, they are also listed in the build section of meta.yaml :eyes: 
* [x] No activate scripts                                                            
* [x] Not a dependency of conda

With one exception, the `entry_points` are *not* listed, yet, so to match cylc-flow which doesn't list `entry_points`, however, still appears to work.

After a bit of digging, I think that:

1) Only `console_scripts` need be listed in `entry_points` (not all `entry_points` as the name suggests).
2) This only matters for Windows where the entry points must be listed to allow the Python executable path to be modified. (https://github.com/conda/conda-build/issues/ 3965)

Because we don't, and won't support Windows, I think we're ok to omit the entry points but would like to double check that.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,8 +30,9 @@ outputs:
       script: python -m pip install . --no-deps --ignore-installed -vv
     requirements:
       host:
+        - __unix
+        - python >=3.7
         - pip
-        - python
       run:
         - python
```
Comment (MetRonnie) [inline recipe/meta.yaml:37]: I think this is needed in every `requirements > X` section (that was what was done in https://github.com/conda-forge/cylc-flow-feedstock/pull/33)
```suggestion
        - __unix
        - python >=3.7
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -28,8 +30,9 @@ outputs:
       script: python -m pip install . --no-deps --ignore-installed -vv
```
Comment (MetRonnie) [inline recipe/meta.yaml:30]: I think this is needed in every `build` section
```suggestion
      script: python -m pip install . --no-deps --ignore-installed -vv
      noarch: python
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -56,8 +60,9 @@ outputs:
       script: python -m pip install .[hub] --no-deps --ignore-installed -vv
```
Comment (MetRonnie) [inline recipe/meta.yaml:60]:
```suggestion
      script: python -m pip install .[hub] --no-deps --ignore-installed -vv
      noarch: python
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -56,8 +60,9 @@ outputs:
...
     requirements:
       host:
+        - __unix
+        - python >=3.7
         - pip
-        - python
       run:
         - {{ pin_subpackage(name + "-base", exact=True) }}
```
Comment (MetRonnie) [inline recipe/meta.yaml:67]: Not sure about this one
```suggestion
        - __unix
        - {{ pin_subpackage(name + "-base", exact=True) }}
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -77,8 +84,9 @@ outputs:
       script: python -m pip install .[hub] --no-deps --ignore-installed -vv
```
Comment (MetRonnie) [inline recipe/meta.yaml:84]:
```suggestion
      script: python -m pip install .[hub] --no-deps --ignore-installed -vv
      noarch: python
```

File: recipe/meta.yaml [hunk 6]
```diff
@@ -77,8 +84,9 @@ outputs:
...
     requirements:
       host:
+        - __unix
+        - python >=3.7
         - pip
-        - python
       run:
         - {{ pin_subpackage(name + "-base", exact=True) }}
```
Comment (MetRonnie) [inline recipe/meta.yaml:91]: Not sure about this one
```suggestion
        - __unix
        - {{ pin_subpackage(name + "-base", exact=True) }}
```

File: recipe/meta.yaml (hunk 2 above)
Comment (oliver-sanders) [reply recipe/meta.yaml:30]: Yeah, took a look at a couple of feedstocks and it looks like it doesn't inherit, ok...

File: recipe/meta.yaml (hunk 4 above)
Comment (oliver-sanders) [reply recipe/meta.yaml:67]: I think we only need to add this to the `base` package as the `pin_subpackage` bit effectively pulls the deps in.

**Comment by MetRonnie** [comment]
@conda-forge-admin, please rerender

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**